tools/remap/convert_remap_actions_to_10x

```diff
@@ -0,0 +1,172 @@
+
+    :param input_line: The line to convert.
+    :return: The converted line.
+
+    :examples:
+        >>> convert_line('# nothing to convert')
+        '# nothing to convert'
+        >>> convert_line('map http://example.com http://backend.example.com')
+        'map http://example.com http://backend.example.com'
+
+        >>> convert_line('map http://example.com http://backend.example.com @action=allow @method=GET')
+        'map http://example.com http://backend.example.com @action=add_allow @method=GET'
+        >>> convert_line('map http://example.com http://backend.example.com @action=deny @method=PUT')
+        'map http://example.com http://backend.example.com @action=add_deny @method=PUT'
+
+        # Verify that add_allow and add_deny are not converted.
+        >>> convert_line('map http://example.com http://backend.example.com @action=add_deny @method=PUT')
+        'map http://example.com http://backend.example.com @action=add_deny @method=PUT'
+        >>> convert_line('map http://example.com http://backend.example.com @action=add_allow @method=GET')
+        'map http://example.com http://backend.example.com @action=add_allow @method=GET'
+
+        # Comments should be converted too.
+        >>> convert_line('# Using @action=allow is nice.')
+        '# Using @action=add_allow is nice.'
+        >>> convert_line('# map http://example.com http://backend.example.com @action=allow @method=GET')
+        '# map http://example.com http://backend.example.com @action=add_allow @method=GET'
+    '''
+    output_line = input_line.replace('@action=allow', '@action=add_allow')
+    output_line = output_line.replace('@action=deny', '@action=add_deny')
+    return output_line
+
+
+def copy_input_file(input: str, no_backup: bool) -> str:
+    '''Create a copy of the input file.
+
+    :param input: The input file path from which to create a copy.
+
+    :param no_backup: Do not create a backup file from input. Rather simply
+    create a temporary file that will be later removed. Otherwise, create a
+    backup file of the format <input>.action_to_add.bak.<num>, where <num> is
+    one greater than the greatest value in the directory where @a input exists.
+
+    :return: The path to the copied file.
+    '''
+    if no_backup:
+        copy_path = tempfile.NamedTemporaryFile(delete=False).name
+    else:
+        input_dir = os.path.dirname(input)
+        input_dir = '.' if input_dir == '' else input_dir
+        input_base = os.path.basename(input)
+        # Find the largest <num> in input_dir/input_base.action_to_add.bak.<num>.
+        max_num = 0
+        for file in os.listdir(input_dir):
+            if file.startswith(f'{input_base}.action_to_add.bak.'):
+                try:
+                    num = int(file.split('.')[-1])
+                except ValueError:
+                    # The suffix is not a number. Ignore this file.
+                    continue
```

Comment:
Testing `if num.isdigit()` would be shorter and more explicit. No need for try/except here.

Reply:
Cool.